main.js

```diff
@@ -28,10 +30,27 @@ class NonCacheRewriter {
     } else {
       this.nativeRewriter = new DummyRewriter()
     }
+    if (config?.orchestrion) {
+      const { instrumentations } = yaml.load(config.orchestrion)
+      this.orchestrionModules = new Set(instrumentations.map((i) => i.module_name))
+    }
   }
 
-  rewrite (code, file) {
-    const response = this.nativeRewriter.rewrite(code, file)
+  rewrite (code, file, passes) {
+    let moduleName
+    let moduleVersion
+    if (passes.includes('orchestrion')) {
+      const details = getNameAndVersion(file)
+      moduleName = details.name
+      moduleVersion = details.version
+      if (!this.orchestrionModules.has(moduleName)) {
+        passes.splice(passes.indexOf('orchestrion'), 1)
```

Comment:
Nit: the indexOf can be moved up before the outer if to prevent the includes call.